.circleci/config.yml

```diff
@@ -2062,16 +2064,101 @@ jobs:
       - run:
           name: Build kona host
           command: just build-kona-host
-          working_directory: kona
+          working_directory: kona-proofs
       - save_cache:
-          key: kona-host-{{ checksum "./kona/justfile" }}-{{ checksum "./kona/version.json" }}
+          key: kona-host-{{ checksum "./kona-proofs/justfile" }}-{{ checksum "./kona-proofs/version.json" }}
           name: Save Kona host to cache
           paths:
-            - "kona/bin/kona-host"
+            - "kona-proofs/bin/kona-host"
       - persist_to_workspace:
           root: .
           paths:
-            - "kona/bin/kona-host"
+            - "kona-proofs/bin/kona-host"
+
+  # Build a single Rust binary for sysgo tests (parameterized).
+  rust-binary-build:
+    docker:
+      - image: <<pipeline.parameters.default_docker_image>>
+    resource_class: xlarge
+    parameters:
+      cache_name:
+        description: "Identifier for cache key and logging"
+        type: string
+      directory:
+        description: "Directory containing the Cargo workspace"
+        type: string
+      build_command:
+        description: "Cargo build command to run"
+        type: string
+      check_binary:
+        description: "Binary name to check for cache hit (in target/release/)"
+        type: string
+      needs_clang:
+        description: "Whether to install clang (needed by bindgen for reth-mdbx-sys)"
+        type: boolean
+        default: false
+    steps:
+      - utils/checkout-with-mise:
+          checkout-method: blobless
+      - run:
+          name: Initialize << parameters.directory >> submodule (shallow)
+          command: git submodule update --init --recursive --depth 1 << parameters.directory >>
```

Comment:
Would be good to also only run this step if there's a cache miss. IIUC it's not necessary at all to checkout the submodule if we hit a cached binary. E.g. for `rust-build-kona` this checkout alone takes 33s and recursively checks out a lot of submodules, which are completely unused.